From 636268657104361edaa9cec919df6318765c2e23 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 9 Dec 2016 02:12:50 +0000
Subject: [PATCH 3/3] truetype: Only enable subpixel rendering when a specific
 file exists

This file is provided by the freetype2-subpixel package in Solus,
and without it installed, subpixel rendering is completely disabled
and we fall back to TT_INTERPRETER_VERSION_35.

With it enabled, and the previous patches, we have the options of both
the Infinality driver (Env vars toggle), but we default to the new
minimal TTF driver.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/truetype/ttobjs.c | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/src/truetype/ttobjs.c b/src/truetype/ttobjs.c
index e4775a5..dad7769 100644
--- a/src/truetype/ttobjs.c
+++ b/src/truetype/ttobjs.c
@@ -36,6 +36,8 @@
 #include "ttgxvar.h"
 #endif
 
+#include <unistd.h>
+
   /**************************************************************************
    *
    * The macro FT_COMPONENT is used in trace mode.  It is an implicit
@@ -1415,12 +1417,24 @@
 
     TT_Driver  driver = (TT_Driver)ttdriver;
 
+    FT_Bool enable_subpixel = FALSE;
+
+    /* If the file path exists, allow the use of subpixel rendering. This
+     * allows the library to be built with the functionality, but also to
+     * disable the functionality with the removal of the package.
+     */
+#ifdef TT_CONFIG_OPTION_SUBPIXEL_HINTING
+    if (access("/usr/share/fonts/enable_subpixel_rendering", F_OK) == 0) {
+        enable_subpixel = TRUE;
+    }
+#endif /* !TT_CONFIG_OPTION_SUBPIXEL_HINTING */
+
     driver->interpreter_version = TT_INTERPRETER_VERSION_35;
 #ifdef TT_SUPPORT_SUBPIXEL_HINTING_INFINALITY
-    driver->interpreter_version = TT_INTERPRETER_VERSION_38;
+    driver->interpreter_version = enable_subpixel == TRUE ? TT_INTERPRETER_VERSION_38 : TT_INTERPRETER_VERSION_35;
 #endif
 #ifdef TT_SUPPORT_SUBPIXEL_HINTING_MINIMAL
-    driver->interpreter_version = TT_INTERPRETER_VERSION_40;
+    driver->interpreter_version = enable_subpixel == TRUE ? TT_INTERPRETER_VERSION_40 : TT_INTERPRETER_VERSION_35;
 #endif
 
 #else /* !TT_USE_BYTECODE_INTERPRETER */
-- 
2.24.1

